pyomo/common/collections/orderedset.py

```diff
@@ -9,42 +9,30 @@
 #  This software is distributed under the 3-clause BSD License.
 #  ___________________________________________________________________________
 
-from collections.abc import MutableSet
 from collections import OrderedDict
+from collections.abc import MutableSet
+from pyomo.common.autoslots import AutoSlots
 
 
-class OrderedSet(MutableSet):
+class OrderedSet(AutoSlots.Mixin, MutableSet):
```

Comment:
Aren't ComponentSets themselves ordered, now the dicts are ordered? This class seems like it's becoming a moot point, unless I'm misunderstanding something. I guess it's doesn't have the overhead of checking for hashability?

Reply:
They are, and yes: the difference between `ComponentSet` and `OrderedSet` is very small (really just the overhead for hashability) - that said, the hasher is not trivial, so there is a performance difference.  The other reason to keep them both is backwards compatibility (and there is a nice symmetry to have an `OrderedSet` to go with `OrderedDict`)